packages/next/client/index.js

```diff
@@ -309,6 +309,14 @@ export async function render(renderingProps) {
   }
 
   try {
+    if (!('IntersectionObserver' in window)) {
+      // This is practically just Safari 11 and 12.0
+      /**
+       * If we find any more such discrepencies between script type=module browsers
+       * then this can be extended as a separate bundle to address them.
+       */
+      await import('next/dist/build/polyfills/intersection-observer')
```

Comment:
It would be good to hoist this import to ~L23 so it starts as early as possible, then await its Promise here:

```js
const withIntersectionObserver = 'IntersectionObserver' in window || import('next/dist/build/polyfills/intersection-observer').catch(e => console.log(e))

// ...

    try {
      await withIntersectionObserver
      await doRender(renderingProps)
    }
```

Reply:
If you do that, I guess you wouldn't want it to render if it fails to load?